### PR TITLE
[fga] Introduce sshkeyservice

### PR DESCRIPTION
**Pull request by jeanp413**
## Description

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at b816b9d</samp>

This pull request introduces SSH key management functionality to the server component, by adding a new SSHKeyService class that handles the logic of creating, updating, and deleting SSH keys for users and workspaces. It also adds unit tests for the service class and refactors the GitpodServerImpl class to use the service methods.

</details>

## Related Issue(s)

Fixes https://linear.app/gitpod/issue/EXP-394/

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/server/src/user/sshkey-service.ts
Comment (jeanp413) [inline components/server/src/user/sshkey-service.ts:1]: Not sure if I should check permissions with `checkPermissionOnUser` as currently only the same user can modify its own ssh keys 🤔

Comment (akosyakov) [reply components/server/src/user/sshkey-service.ts:1]: Yeah, you will need to add permissions checks. I'm not sure which though. cc @svenefftinge

Comment (svenefftinge) [reply components/server/src/user/sshkey-service.ts:1]: We could either use `read_info` and `write_info` for this or introduce two new permissions `read_ssh` and  `write_ssh`. I believe making this explicit is better, so let's do the latter, i.e. enhance the schema.yaml with

two permissions on `user`:

```
   permission `read_ssh` = self
   permission `write_ssh`= self
```

Comment (jeanp413) [reply components/server/src/user/sshkey-service.ts:1]: Updated

**Review by svenefftinge — APPROVED** [review]
Thanks @jeanp413!
I have added one commit that adds an additional requestorId to the service methods. In this case we don't yet have clients where that is not the same as the userId owning the sshkeys, but we apply this pattern everywhere (first arg is the requestor) to be able to allow other users (e.g. admins) to access any resources.
In the other services we have so far named the first arg ` userId` but that is confusing when we are talking about the user a s a resources, so I opted for changing this to `requestorId`. We should go through the other services and rename the first argument from `userId` -> `requestorId` as well.

**Pull request merged**